python/pacemaker/exitstatus.py

```diff
@@ -60,3 +75,9 @@ class ExitStatus(IntEnum):
     DEGRADED_PROMOTED = 191
     NONE = 193
     MAX = 255
+
+
+def exit_str(status):
+    """Given an ExitStatus, return the matching error string."""
+    _libcrmcommon.crm_exit_str.restype = ctypes.c_char_p
+    return _libcrmcommon.crm_exit_str(status).decode()
```

Comment:
It's especially worth thinking about this patch for a couple reasons:

1.  It's going to be public API, so we had better be happy with the name.
2. We're likely to use this same method in the future for adding additional bindings.

Reply:
Yeah, definitely.

Adding this patch on top of the original one (and I'd squash these together):
```
--- a/python/pacemaker/exitstatus.py
+++ b/python/pacemaker/exitstatus.py
@@ -1,6 +1,6 @@
 """A module providing constants relating to why a process or function exited."""
 
-__all__ = ["ExitStatus", "exit_str"]
+__all__ = ["ExitStatus"]
 __copyright__ = "Copyright 2023-2024 the Pacemaker project contributors"
 __license__ = "GNU Lesser General Public License version 2.1 or later (LGPLv2.1+)"
 
@@ -20,6 +20,7 @@ def _load_library(basename):
 
 
 _libcrmcommon = _load_library("crmcommon")
+_libcrmcommon.crm_exit_str.restype = ctypes.c_char_p
 
 
 # These values must be kept in sync with include/crm/common/results.h
@@ -76,8 +77,6 @@ class ExitStatus(IntEnum):
     NONE = 193
     MAX = 255
 
-
-def exit_str(status):
-    """Given an ExitStatus, return the matching error string."""
-    _libcrmcommon.crm_exit_str.restype = ctypes.c_char_p
-    return _libcrmcommon.crm_exit_str(status).decode()
+    def __str__(self):
+        """Given an ExitStatus, return the matching error string."""
+        return _libcrmcommon.crm_exit_str(self.value).decode()
```

Lets us do:
```
$ LD_LIBRARY_PATH=lib/common/.libs PYTHONPATH=python python
Python 3.11.6 (main, Oct  3 2023, 00:00:00) [GCC 13.2.1 20230728 (Red Hat 13.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pacemaker.exitstatus import *
>>> str(ExitStatus.INVALID_PARAM)
'Invalid parameter'
>>> str(ExitStatus.SOFTWARE)
'Internal software bug'
>>> 
```

Reply:
nice